### PR TITLE
Adding an option to block local ansible connections

### PR DESCRIPTION
**Pull request by brenton**

### CODE REVIEW

**Comment by brenton** [comment]
@aweiteka, this is mostly an idea so let me know what you think.  The basic idea is that there are cases in OpenShift where admins installing the product use local ansible connections.  They may feed those inventories in to a tool that uses playbook2image.  If they do that ansible will get really confused.

File: .s2i/bin/run [hunk 1]
```diff
@@ -11,16 +11,19 @@
 # SOURCE and HOME DIRECTORY: /opt/app-root/src
 
 #set -x
+INVENTORY="tmp_inventory"
+INVENTORY_ARG="-i ${INVENTORY}"
```
Comment (brenton) [inline .s2i/bin/run:15]: @aweiteka, I don't feel strongly about these refactorings I just needed some way to consistently reference the non-dynamic inventories.  I certainly didn't want to do `sed -i` on the inventory that was passed in.

Comment (brenton) [reply .s2i/bin/run:15]: Also, the reason I named it `tmp_inventory` was odd.  I need to dig more but when used playbook2image on an inventory in openshift-ansible I soon realized we had a directory called `inventory` that was getting baked in to our image.  I feel like I've seen a few repos with that convention so I renamed it for lack of a better idea.  If we're using playbook2image incorrectly feel free to say so.

Comment (aweiteka) [reply .s2i/bin/run:15]: No issue with tmp_inventory.

**Review by aweiteka — COMMENTED** [review]
I'm trying to understand the case this protects against. Could you describe the situation a bit more? Maybe a more specific hypothetical?

**Comment by brenton** [comment]
@aweiteka, We're writing a playbook that is going to run health checks on an OpenShift host.  We find it common that admins run the openshift installer from one of the hosts where they are actually going to run OpenShift.  When they do that the best thing to do is to set `ansible_connection=local` to avoid the ssh overhead.  If they then reuse that inventory with a image built from playbook2image ansible will get confused.  In our case, instead of ssh'ing to the host and running the checks there it will actually run the checks from inside the image itself which leads to confusing errors.

I can't think of any time someone using playbook2image would want to use `ansible_connection=local` so I was tempted to suggest disabling it by default and perhaps print a warning message of some sort.

**Comment by aweiteka** [comment]
> I can't think of any time someone using playbook2image would want to use ansible_connection=local so I was tempted to suggest disabling it by default and perhaps print a warning message of some sort.

Ah, I see. Makes sense.

**Review by aweiteka — APPROVED** [review]

**Comment by brenton** [comment]
Let me know once you've published a new image!

**Comment by aweiteka** [comment]
@brenton image pushed. Thanks!

**Pull request merged**